TypeCobol/Compiler/Parser/DiagnosticUtils.cs

```diff
@@ -18,9 +18,17 @@ internal static class DiagnosticUtils
             var parserDiag = new ParserDiagnostic(message, e.StartIndex + 1, e.StopIndex + 1, e.ConsumedTokens[0].Line, null, code);
             e.Diagnostics.Add(parserDiag);    
         }
-		internal static void AddError(CodeElement e, string message, Scanner.Token token, string rulestack = null, MessageCode code = MessageCode.SyntaxErrorInParser) {
+		internal static void AddError(CodeElement e, string message, Scanner.Token token, string rulestack = null, MessageCode code = MessageCode.SyntaxErrorInParser, bool withStackRule = true) {
```

Comment:
Could youcheck the behavior when you call the roiginal method with rulestack set to null and no modification. I'm not sure any modification is required.

Reply:
done in last commit